### ISSUE TITLE
RequestCredentials flag withCredentials is not applied

### ISSUE THREAD

**Issue by cliren** [aws-amplify/amplify-js#10789]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

GraphQL API

### Amplify Categories

auth

### Environment information

<details>

```
System:
    OS: macOS 12.3.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 112.14 MB / 32.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 16.17.1 - ~/.nvm/versions/node/v16.17.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.17.1/bin/yarn
    npm: 8.15.0 - ~/.nvm/versions/node/v16.17.1/bin/npm
  Browsers:
    Chrome: 108.0.5359.124
    Firefox: 102.5.0
    Safari: 15.4
  npmPackages:
    @aws-sdk/types: 3.215.0 => 3.215.0 (1.0.0-rc.10, 3.6.1, 3.186.0)
    @commitlint/cli: ^17.1.2 => 17.2.0 
    @commitlint/config-conventional: ^17.1.0 => 17.3.0 
    @commitlint/config-lerna-scopes: ^17.0.2 => 17.2.1 
    fs-extra: 10.1.0 => 10.1.0 (9.1.0, 8.1.0, 1.0.0)
    husky: ^8.0.1 => 8.0.2 
    lerna: ^5.6.2 => 5.6.2 (6.0.1)
  npmGlobalPackages:
    corepack: 0.12.1
    generator-code: 1.7.0
    npm: 8.15.0
    ts-node: 10.9.1
    typescript: 4.8.4
    uuid: 9.0.0
    uuid4: 2.0.3
    yarn: 1.22.19
    yo: 4.3.0


```

</details>

### Describe the bug

Following is invoked on a button click in a react app, `withCredentials: true` is set but cookie is not sent. Request is cross domain and all required CORS steps were taken (required CORS headers were set, cookie is set to Secure:true, SameSite: None and set on the same domain as the react app).

### Not working

Using @aws-amplify/api-graphql

```
import {GraphQLAPIClass, GraphQLResult} from '@aws-amplify/api-graphql';
...

export function createAmplifyAPI(): GraphQLAPIClass {
    const graphqlApi = new GraphQLAPIClass({
        withCredentials: true
    });

    graphqlApi.configure({
        aws_appsync_region: 'us-west-2',
        aws_appsync_authenticationType: 'AWS_LAMBDA',
        aws_appsync_graphqlEndpoint: appSyncEndpoint
    });
    return graphqlApi;
}

const executeAmplifyJS = async <TQuery, TVars extends GQLVariables>(
    mutationOptions: MutationOptions<TQuery, TVars>
) => {
    const graphqlApi = createAmplifyAPI();
    const { mutation, variables } = mutationOptions;
    const response = graphqlApi.graphql<TQuery>(
        {
            query: mutation,
            variables: variables,
            authToken: authToken
        }
    ) as Promise<GraphQLResult<TQuery>>;
    return response;
};

export const testAmplifyJS = async () => {
    const mutation = `mutation MyMutation {
      createPost(id: "test")
    }
    `;
    const parsedMutation = parse(mutation);

    const res = await executeAmplifyJS({ mutation: parsedMutation });
    console.log('GraphQL Response: ', res);
};

```

###  Working

Using another library

```
import { GraphQLClient } from 'graphql-request';
...

const gqlRequestClient = new GraphQLClient(appSyncEndpoint, {
    credentials: 'include',
    headers: {
        Authorization: 'test-token'
    }
});

const executeGraphQLRequest = async <TQuery, TVars extends GQLVariables>(
    mutationOptions: MutationOptions<TQuery, TVars>
) => {
    const { mutation, variables } = mutationOptions;
    const requestHeaders = {
        authorization: authToken
    };

    const response = await gqlRequestClient.request(mutation, variables, requestHeaders);
    return response;
};

export const testGraphQLRequest = async () => {
    const mutation = `mutation MyMutation {
          createPost(id: "test")
        }
        `;
    const parsedMutation = parse(mutation);

    const res = await executeGraphQLRequest({ mutation: parsedMutation });
    console.log('GraphQL Response: ', res);
};

```

### Working - using rest-api

Using '@aws-amplify/api-rest' works fine. There is an issue in api-graphql that's not passing `withCredentials` to api-rest api.

```
import { RestClient } from '@aws-amplify/api-rest';

export const testRestClient = async () => {
    const apiOptions = {
        headers: {
            Authorization: 'test-token'
        },
        endpoints: [
            {
                name: 'myApi',
                endpoint: appSyncEndpoint
            },
        ],
    };

    const graphQLRequest = '{"query":"mutation MyMutation {\\n  createPost(id: \\"test\\")\\n}\\n","operationName":"MyMutation"}';

    const restClient = new RestClient(apiOptions);
    const res = await restClient.ajax(appSyncEndpoint, "POST",{
        withCredentials: true,
        body: graphQLRequest
    })
    console.log('GraphQL Response: ', res);
};

```

### Expected behavior

In the above example, cookie is sent using another graphql client but not using amplifyJS. Expect `withCredentials` flag is passed to underling rest-client (using axios).

### Reproduction steps

Pasted code above in description, don't have an active github repo to demonstrate

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by dpilch** [comment]
Hi @cliren, Amplify creates a singleton of the GraphQLAPI. Please use this instead of creating a new instance.

```
import { GraphQLAPI } from '@aws-amplify/api-graphql';
// alternatively 
// import { API } from 'aws-amplify';

...


GraphQLAPI.graphql(...)
// alternatively 
// API.graphql(...)
```

Please let us know if this resolves your issue.

**Comment by cliren** [comment]
@dpilch Our use case is, we have a network client that uses `@aws-amplify/api-graphql` to talk to AppSync endpoint. We don't use AmplifyCLI directly. Do we still need to import `aws-amplify`?

**Comment by chrisbonifacio** [comment]
Hi @cliren we merged a feature PR to allow `withCredentials` on graphql requests. Has this resolved the issue for you on the latest version?

Also, to answer your previous question you don't have to import `aws-amplify` if you're getting `GraphQLAPI.graphql` from `@aws-amplify/api-graphql`.

to rephrase Dane's code, you can do either

```js
import { GraphQLAPI } from '@aws-amplify/api-graphql';
...
GraphQLAPI.graphql(...)

```

or

```js
import { API } from 'aws-amplify';
...
API.graphql(...)
```

**Comment by dpilch** [comment]
The `withCredentials` has been released in `aws-amplify@5.0.12`.
